lib/livebook/utils/ansi.ex

```diff
@@ -31,14 +33,22 @@ defmodule Livebook.Utils.ANSI do
   @doc """
   Takes a string with ANSI escape codes and parses it
   into a list of `{modifiers, string}` parts.
+
+  Also returns the final modifiers.
+
+  ## Options
+
+    * `:modifiers` - a list with initial modifiers
   """
-  @spec parse_ansi_string(String.t()) :: list({list(modifier()), String.t()})
-  def parse_ansi_string(string) do
+  @spec parse_ansi_string(String.t(), keyword()) :: {list({modifiers(), String.t()}), modifiers()}
+  def parse_ansi_string(string, opts \\ []) do
+    modifiers = opts |> Keyword.get(:modifiers, []) |> Map.new()
```

Comment:
Maybe keep it as a map on input and output to avoid the conversion?

Reply:
I think logically a list of entries makes more sense, we use map internally mostly for this:

https://github.com/livebook-dev/livebook/blob/aee6d5ab52b7a28e468af6e4449cd339a693137b/lib/livebook/utils/ansi.ex#L240-L243

But since this map will have at most 5 entries, we could as well use `Keyword`, right?

Reply:
They are key-values though, right?

Reply:
Correct

https://github.com/livebook-dev/livebook/blob/d5c6dfadcb993374aca555e710938629d772663a/lib/livebook/utils/ansi.ex#L4-L9